app/controllers/posts_controller.rb

```diff
@@ -5,7 +5,32 @@ def index
   end
 
   def show
+    @user = ApplicationController.current_user
     @post = Post.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = ApplicationController.current_user
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      flash[:success] = 'Post created successfully'
+      redirect_to user_posts_path(ApplicationController.current_user)
+    else
+      flash[:error] = 'Error: Post could not be saved'
+      render 'new'
+    end
+  end
+
```

Comment:
- [ ] Great job working on the functionality for adding a new post and a new comment but I just noticed that you missed to add functionality for adding a like so please make sure to add a controller to create a new like and a specific routes to it and please don't forget to add a button in the views to add a new like in specific post  as required in the requirements 🤩💪🏻💯 

![image](https://user-images.githubusercontent.com/58553711/230529702-ad06405d-a063-46dd-b070-090e4b10bdf6.png)